src/sinks/clickhouse/integration_tests.rs

```diff
@@ -324,7 +324,7 @@ async fn no_retry_on_incorrect_data_warp() {
         .unwrap()
         .unwrap();
 
-    assert_eq!(receiver.try_recv(), Ok(BatchStatus::Rejected));
+    assert_eq!(receiver.try_recv(), Ok(BatchStatus::Errored));
```

Comment:
As clickhouse now uses the generic HttpService, the `Error` BatchStatus is now returned on server errors

Reply:
Hmm so this PR is introducing a behavior change to the case of 500-level HTTP error codes for the sink.

We used to consider them a non-recoverable failure, and the behavior now (with BatchStatus::Errored) is that they are considered retriable.

Since we moved to a direction from not retrying to retrying, I would consider this a non-breaking change. If we decide to keep this change, then I think it would deserve a mention in the changelog fragments as an enhancement.

However, what I don't know offhand, is if the Clickhouse service is such that we very specifically do not want to retry on 500 level errors due to the behavior of that service.

I'd like to open this up to thoughts from @bruceg , as you probably have a better understanding of the behavioral implications to the batch status changing, for example to event finalizers etc.

Reply:
I'm currently looking into how we can preserve the custom `ClickhouseResponse` implementation while using the http service...

Reply:
I don't know of any reason to treat Clickhouse specially for errors, though I have less experience with the integrations side of things. I do agree this would be a behavior change that warrants a changelog entry, though.

WRT event finalization, though, `Errored` and `Rejected` are treated effectively equivalent. The only difference would be the end status.

Reply:
Sorry, the request retry behavior is unchanged. Don't mind me :P  

@bruceg  check my understanding- the `Errored` end status, that means that the batch would be able to be re-tried in e2e acks , true?

Reply:
Correct.

Reply:
> For context, the only reason ClickHouse had special behavior was because of [this legacy behavior](https://github.com/vectordotdev/vector/blob/53f97c1c61ca176ba20852d0cfc1e45e44cf2235/src/sinks/clickhouse/service.rs#L87), which has been fixed since 2019.

Ahhh, good to know.